src/Mod/TechDraw/TechDrawTools/TaskHoleShaftFit.py

```diff
@@ -0,0 +1,183 @@
+# ***************************************************************************
+# *   Copyright (c) 2023 edi <edi271@a1.net>                                *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+"""Provides the TechDraw HoleShaftFit Task Dialog."""
+
+__title__ = "TechDrawTools.TaskHoleShaftFit"
+__author__ = "edi"
+__url__ = "https://www.freecadweb.org"
+__version__ = "00.01"
+__date__ = "2023/02/07"
+
+from PySide.QtCore import QT_TRANSLATE_NOOP
+from PySide import QtCore
+import PySide.QtGui as QtGui
+
+import FreeCAD as App
+import FreeCADGui as Gui
+
+from functools import partial
+
+import os
+
+class TaskHoleShaftFit:
+    def __init__(self,sel):
+
+        self.isHole = True
+        self.sel = sel
+        self.holeValues = [["h9","D10","loose"],["h9","E9","loose"],["h9","F8","Loose"],["h6","G7","loose"],
```

Comment:
`["h9","D10","loose"]` What does each of these values mean? I don't know anything about ISO holes... ELI5

Reply:
https://www.cobanengineering.com/Tolerances/ISOShaftTolerances3mmTo400mm.asp

First column in the chart is the source of the "magic numbers".

Reply:
Nothing can be produced exact. Therefore ISO286 defines quality fields. Assume you have to produce several shafts. The shafts do not need to have the exact diameter, their diameter must be inside a defined field.
For the following example see table posted above by Wandererfan:
Assume the specified size of the shafts diameter is 35mm. If our field is f7, all diameters must be less then the specified size. They have to be between minimum 25 micrometers, maximum 50 micrometers less. Maybe our shafts are the axes of rotating parts.
Lower case fields define shafts, upper case fields define holes. The bigger the number, the bigger the field range, compare h10, h11, h12. Letters a to g define undersize, j to z define oversize. The h and H fields are special fields, one value is 0.
There are always two parts combined, one shaft and one hole. Three types of fit are possible: loose fit, snug fit and press fit.
["h9","D10","loose"] means:
Shaft has field h quality 9. Hole has field D, quality 10. The type of fit is loose, a moving between the parts is possible. In all combinations one of the two parts has the h (or H) field.

Reply:
For some reason I didn't get a notification of your reply. Idk if it's just for me it's not intuitive, so I would probably make it a dictionary or smth like that instead, so what the variable values mean are very clear